### PR TITLE
Use a strong type for upstairs negotiation

### PR DESCRIPTION
**Pull request by mkeeter**
This should be equivalent, but is easier to read.

### CODE REVIEW

File: upstairs/src/lib.rs [hunk 1]
```diff
@@ -1162,28 +1178,30 @@ where
...
-     *
+     * NegotiationState::GetLastFlush (offline only)
+     * ------------------------------
      *          Upstairs             Downstairs
-     * 3:       LastFlush(lf)) --->
+     *          LastFlush(lf)) --->
      *                         <---  LastFlushAck(lf)
      *
      * After receiving our last flush, we now move this downstairs state to
      * Replay and skip ahead to step 5.
```
Comment (leftwo) [inline upstairs/src/lib.rs:1188]: Step 5 is now `NegotiationState::Done`

File: upstairs/src/lib.rs [hunk 2]
```diff
@@ -1162,28 +1178,30 @@ where
      * also request extent versions and will have to repair this
      * downstairs, skipping over step 3 as well.
      *
-     * (Offline only):
-     *
+     * NegotiationState::GetLastFlush (offline only)
```
Comment (leftwo) [inline upstairs/src/lib.rs:1181]: The comment above here should be updated to replace `3` and `4` with 
`NegotiationState::GetLastFlush` and `NegotiationState::GetExtentVersions`

**Review by leftwo — APPROVED** [review]
Just some comment cleanup

File: upstairs/src/lib.rs (hunk 1 above)
Comment (mkeeter) [reply upstairs/src/lib.rs]: Done in 97dfe4778518d3536ecc2adb75bf2b9672296416

File: upstairs/src/lib.rs (hunk 2 above)
Comment (mkeeter) [reply upstairs/src/lib.rs:1184]: Done in 97dfe4778518d3536ecc2adb75bf2b9672296416

**Pull request merged**